flink-java/src/test/java/org/apache/flink/api/java/io/PrintingOutputFormatTest.java

```diff
@@ -0,0 +1,130 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.java.io;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Tests for the {@link PrintingOutputFormat}.
+ */
+public class PrintingOutputFormatTest {
+
+	private final PrintStream originalSystemOut = System.out;
+	private final PrintStream originalSystemErr = System.err;
+
+	private final ByteArrayOutputStream arrayOutputStream = new ByteArrayOutputStream();
+	private final ByteArrayOutputStream arrayErrorStream = new ByteArrayOutputStream();
+
+	private final String line = System.lineSeparator();
+
+	@Before
+	public void setUp() {
+		System.setOut(new PrintStream(arrayOutputStream));
+		System.setErr(new PrintStream(arrayErrorStream));
+	}
+
+	@After
+	public void tearDown() {
+		if (System.out != originalSystemOut) {
+			System.out.close();
+		}
+		if (System.err != originalSystemErr) {
+			System.err.close();
+		}
+		System.setOut(originalSystemOut);
+		System.setErr(originalSystemErr);
+	}
+
+	@Test
+	public void testPrintOutputFormatStdOut() throws Exception {
+		PrintingOutputFormat<String> printSink = new PrintingOutputFormat<>();
+		printSink.open(0, 1);
+
+		printSink.writeRecord("hello world!");
+
+		assertEquals("Print to System.out", printSink.toString());
+		assertEquals("hello world!" + line, arrayOutputStream.toString());
+		printSink.close();
+	}
+
+	@Test
+	public void testPrintOutputFormatStdErr() throws Exception {
+		PrintingOutputFormat<String> printSink = new PrintingOutputFormat<>(true);
+		printSink.open(0, 1);
+
+		printSink.writeRecord("hello world!");
+
+		assertEquals("Print to System.err", printSink.toString());
+		assertEquals("hello world!" + line, arrayErrorStream.toString());
+		printSink.close();
+	}
+
+	@Test
+	public void testPrintOutputFormatWithPrefix() throws Exception {
+		PrintingOutputFormat<String> printSink = new PrintingOutputFormat<>();
+		printSink.open(1, 2);
+
+		printSink.writeRecord("hello world!");
+
+		assertEquals("Print to System.out", printSink.toString());
+		assertEquals("2> hello world!" + line, arrayOutputStream.toString());
+		printSink.close();
+	}
+
+	@Test
+	public void testPrintOutputFormatWithIdentifierAndPrefix() throws Exception {
+		PrintingOutputFormat<String> printSink = new PrintingOutputFormat<>("mySink", false);
+		printSink.open(1, 2);
+
+		printSink.writeRecord("hello world!");
+
+		assertEquals("Print to System.out", printSink.toString());
+		assertEquals("mySink:2> hello world!" + line, arrayOutputStream.toString());
+
+		printSink = new PrintingOutputFormat<>("mySink", true);
+		assertEquals("Print to System.err", printSink.toString());
+		assertEquals("mySink:2> hello world!" + line, arrayOutputStream.toString());
```

Comment:
Were you trying in those 3 lines test for stderr? If so, you haven't invoked anything on stderr instance of `printSink` and in this line, you are just asserting again the standard output. For asserting stderr you should check against `arrayErrorStream` instead of `arrayOutputStream`. 

However, I don't think it is necessary here to test for prefix AND for stderr together. So I would be actually in favour of dropping those 3 lines here (If you want you can keep those assertions after fixing the before mentioned bug).